libs/sysplugins/smarty_internal_runtime_updatescope.php

```diff
@@ -44,16 +44,29 @@ public function _updateScope(Smarty_Internal_Template $tpl, $varName, $tagScope
      * Get array of objects which needs to be updated  by given scope value
      *
      * @param Smarty_Internal_Template $tpl
-     * @param int                      $mergedScope merged tag and template scope to which bubble up variable value
+     * @param int                      $tagScope tag scope to which bubble up variable value
      *
      * @return array
      */
-    public function _getAffectedScopes(Smarty_Internal_Template $tpl, $mergedScope)
+    public function _getAffectedScopes(Smarty_Internal_Template $tpl, $tagScope)
     {
+        $mergedScope = $tagScope | $tpl->scope;
         $_stack = array();
         $ptr = $tpl->parent;
         if ($mergedScope && isset($ptr) && $ptr->_isTplObj()) {
             $_stack[] = $ptr;
+            if ($tpl->inheritance && $tagScope & Smarty::SCOPE_PARENT) {
```

Comment:
Templates with an inheritance chain have an implicit `Smarty::SCOPE_PARENT`. I don't know the exact reason but the variables should not bubble up outside the inheritance chain implicitly so I have to explicitly test against the tag scope.